### PR TITLE
[vmware] global reach connections

### PR DESCRIPTION
**Pull request by ctaggart**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
vmware

**Comment by ctaggart** [comment]
Hi @Juliehzl, can I please get a review?

File: src/vmware/azext_vmware/tests/latest/test_global_reach_connection_scenario.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        count = len(self.cmd('az vmware global-reach-connection list -g {rg} -c {privatecloud}').get_output_in_json())
+        self.assertEqual(count, 0, 'count expected to be 0')
+
+        self.cmd('az vmware global-reach-connection create -g {rg} -c {privatecloud} -n {global_reach_connection} --peer-express-route-circuit {peer_express_route_circuit} --authorization-key {authorization_key}')
+
+        count = len(self.cmd('az vmware global-reach-connection list -g {rg} -c {privatecloud}').get_output_in_json())
+        self.assertEqual(count, 1, 'count expected to be 1')
+
+        self.cmd('vmware global-reach-connection show -g {rg} -c {privatecloud} -n {global_reach_connection}')
```
Comment (Juliehzl) [inline src/vmware/azext_vmware/tests/latest/test_global_reach_connection_scenario.py:38]: could you add check for each command?

File: src/vmware/azext_vmware/commands.py [hunk 2]
```diff
@@ -80,3 +81,9 @@ def load_command_table(self, _):
         g.custom_show_command('show', 'addon_srm_show')
         g.custom_command('update', 'addon_srm_update')
         g.custom_command('delete', 'addon_srm_delete')
+
+    with self.command_group('vmware global-reach-connection', vmware_sdk, client_factory=cf_vmware) as g:
+        g.custom_command('create', 'globalreachconnection_create')
+        g.custom_command('list', 'globalreachconnection_list')
+        g.custom_command('delete', 'globalreachconnection_delete')
```
Comment (Juliehzl) [inline src/vmware/azext_vmware/commands.py:88]: what about add confirmation for delete command?

File: src/vmware/azext_vmware/tests/latest/test_global_reach_connection_scenario.py (hunk 1 above)
Comment (ctaggart) [reply src/vmware/azext_vmware/tests/latest/test_global_reach_connection_scenario.py]: I added some asserts.

File: src/vmware/azext_vmware/commands.py (hunk 2 above)
Comment (ctaggart) [reply src/vmware/azext_vmware/commands.py:88]: I'm going to assume no. I've asked the PM. The only delete that confirms right now is `private-cloud`. If the PM comes back with a different answer, I'll put in another PR.

File: src/vmware/CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ Other changes:
 - Add `az vmware addon hcx` command group to manage a HCX addon
 - Add `az vmware addon srm` command group to manage a Site Recovery Manager (SRM) addon
 - Add `az vmware addon vr` command group to manage a vSphere Replication (VR) addon
+- Add `az vmware global-reach-connection` command group to manage Global Reach Connections
```
Comment (Juliehzl) [inline src/vmware/CHANGELOG.md:14]: I see there are many features in 3.0, but not released. Do you still have new features to add? If not, recommend to upgrade version in setup.py here. Or will you raise a new PR to upgrade the version in setup.py tp trigger our extension release pipeline?

Comment (ctaggart) [reply src/vmware/CHANGELOG.md:14]: The new API is still behind a feature flag in production.  After the feature flag is removed, I'll create a new PR to upgrade the version in setup.py to trigger the release.

**Comment by ctaggart** [comment]
@Juliehzl, are you able to merge this? Are there any outstanding concerns?

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**